src/operator/mxnet_op.h

```diff
@@ -88,6 +88,36 @@ struct clip_grad {
   }
 };
 
+struct reverse {
+  MSHADOW_XINLINE static int ReverseIndex(index_t index, int ndim, const index_t* dims,
+                                          const bool* reverse, const index_t* offset) {
+    const index_t ndim_minus = ndim - 1;
```

Comment:
have you checked the speed vs flip? This implementation seems very inefficient. See https://devblogs.nvidia.com/parallelforall/fast-dynamic-indexing-private-arrays-cuda/